### PR TITLE
Package oraft.0.1.0

### PR DESCRIPTION
**Pull request by komamitsu**
### `oraft.0.1.0`
Raft consensus algorithm implemented in OCaml
Raft consensus algorithm implemented in OCaml

---
* Homepage: https://github.com/komamitsu/oraft
* Source repo: git+https://github.com/komamitsu/oraft.git
* Bug tracker: https://github.com/komamitsu/oraft/issues

---
:camel: Pull-request generated by opam-publish v2.0.0

### CODE REVIEW

**Comment by kit-ty-kate** [comment]
Something went wrong with your git setup before calling `opam publish` because the `url` section is not present. Could you try again with the latest version of `opam-publish`?

File: packages/oraft/oraft.0.1.0/opam [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+description: "Raft consensus algorithm implemented in OCaml"
+synopsis: "Raft consensus algorithm implemented in OCaml"
+license: "Apache"
+build: [
+  ["dune" "build" "-p" name "-j" jobs]
+  ["dune" "runtest" "-p" name "-j" jobs] {with-test}
+]
+
+depends: [
+    "dune" {build}
```
Comment (kit-ty-kate) [inline packages/oraft/oraft.0.1.0/opam:16]: Also `dune` should not be tagged as a build dependency and no `dune-project` file is present in your source repository. Could you untag it and add the appropriate version of dune required according to your `dune-project` file?

File: packages/oraft/oraft.0.1.0/opam [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+    "core"
```
Comment (kit-ty-kate) [inline packages/oraft/oraft.0.1.0/opam:17]: Something tells me this is going to require a version constraint somewhere. This library changed in API often in the past so it's a good idea to know which version is required at a minimum (probably something like `>= "v0.9.0"`)

File: packages/oraft/oraft.0.1.0/opam (hunk 1 above)
Comment (komamitsu) [reply packages/oraft/oraft.0.1.0/opam]: Ah, I happened to add `dune-project` to my global gitignore unexpectedly... Just untagged `dune` and added `dune-project` file.

**Comment by komamitsu** [comment]
Thanks for the comments! Addressed them.

**Comment by kit-ty-kate** [comment]
I just fixed two issues in your package description. Could you double-check my fix is right and return it upstream if it is?

**Comment by kit-ty-kate** [comment]
The build also failed on OCaml < 4.05:
```
#=== ERROR while compiling oraft.0.1.0 ========================================#
# context              2.0.6 | linux/x86_64 | ocaml-base-compiler.4.04.2 | pinned(https://github.com/komamitsu/oraft/archive/0.1.0.tar.gz)
# path                 ~/.opam/4.04/.opam-switch/build/oraft.0.1.0
# command              ~/.opam/4.04/bin/dune build -p oraft -j 72
# exit-code            1
# env-file             ~/.opam/log/oraft-23-d3fb42.env
# output-file          ~/.opam/log/oraft-23-d3fb42.out
### output ###
# File "/home/opam/.opam/4.04/lib/ppx_inline_test/ppx_inline_test.dune", line 1, characters 0-0:
# Warning: .dune files are ignored since 2.0. Reinstall the library with dune
# >= 2.0 to get rid of this warning and enable support for the subsystem this
# library provides.
# File "/home/opam/.opam/4.04/lib/ppx_expect/ppx_expect.dune", line 1, characters 0-0:
# Warning: .dune files are ignored since 2.0. Reinstall the library with dune
# >= 2.0 to get rid of this warning and enable support for the subsystem this
# library provides.
#       ocamlc lib/.oraft.objs/byte/oraft__Request_dispatcher.{cmo,cmt} (exit 2)
# (cd _build/default && /home/opam/.opam/4.04/bin/ocamlc.opt -w -40 -g -bin-annot -I lib/.oraft.objs/byte -I /home/opam/.opam/4.04/lib/astring -I /home/opam/.opam/4.04/lib/base -I /home/opam/.opam/4.04/lib/base/caml -I /home/opam/.opam/4.04/lib/base/md5 -I /home/opam/.opam/4.04/lib/base/shadow_stdlib -I /home/opam/.opam/4.04/lib/base64 -I /home/opam/.opam/4.04/lib/bin_prot -I /home/opam/.opam/4.04/lib/bin_prot/shape -I /home/opam/.opam/4.04/lib/biniou -I /home/opam/.opam/4.04/lib/bytes -I /home/opam/.opam/4.04/lib/cohttp -I /home/opam/.opam/4.04/lib/cohttp-lwt -I /home/opam/.opam/4.04/lib/cohttp-lwt-unix -I /home/opam/.opam/4.04/lib/conduit -I /home/opam/.opam/4.04/lib/conduit-lwt -I /home/opam/.opam/4.04/lib/conduit-lwt-unix -I /home/opam/.opam/4.04/lib/core -I /home/opam/.opam/4.04/lib/core_kernel -I /home/opam/.opam/4.04/lib/core_kernel/base_for_tests -I /home/opam/.opam/4.04/lib/domain-name -I /home/opam/.opam/4.04/lib/easy-format -I /home/opam/.opam/4.04/lib/fieldslib -I /home/opam/.opam/4.04/lib/fmt -I /home/opam/.opam/4.04/lib/ipaddr -I /home/opam/.opam/4.04/lib/ipaddr-sexp -I /home/opam/.opam/4.04/lib/ipaddr/unix -I /home/opam/.opam/4.04/lib/jane-street-headers -I /home/opam/.opam/4.04/lib/logs -I /home/opam/.opam/4.04/lib/lwt -I /home/opam/.opam/4.04/lib/lwt/unix -I /home/opam/.opam/4.04/lib/macaddr -I /home/opam/.opam/4.04/lib/magic-mime -I /home/opam/.opam/4.04/lib/mmap -I /home/opam/.opam/4.04/lib/ocaml/threads -I /home/opam/.opam/4.04/lib/ocplib-endian -I /home/opam/.opam/4.04/lib/parsexp -I /home/opam/.opam/4.04/lib/ppx_assert/runtime-lib -I /home/opam/.opam/4.04/lib/ppx_bench/runtime-lib -I /home/opam/.opam/4.04/lib/ppx_compare/runtime-lib -I /home/opam/.opam/4.04/lib/ppx_deriving/runtime -I /home/opam/.opam/4.04/lib/ppx_deriving_yojson/runtime -I /home/opam/.opam/4.04/lib/ppx_expect/collector -I /home/opam/.opam/4.04/lib/ppx_expect/common -I /home/opam/.opam/4.04/lib/ppx_expect/config -I /home/opam/.opam/4.04/lib/ppx_hash/runtime-lib -I /home/opam/.opam/4.04/lib/ppx_inline_test/config -I /home/opam/.opam/4.04/lib/ppx_inline_test/runtime-lib -I /home/opam/.opam/4.04/lib/ppx_sexp_conv/runtime-lib -I /home/opam/.opam/4.04/lib/re -I /home/opam/.opam/4.04/lib/re/posix -I /home/opam/.opam/4.04/lib/result -I /home/opam/.opam/4.04/lib/seq -I /home/opam/.opam/4.04/lib/sexplib -I /home/opam/.opam/4.04/lib/sexplib/unix -I /home/opam/.opam/4.04/lib/sexplib0 -I /home/opam/.opam/4.04/lib/spawn -I /home/opam/.opam/4.04/lib/splittable_random -I /home/opam/.opam/4.04/lib/stdio -I /home/opam/.opam/4.04/lib/stdlib-shims -I /home/opam/.opam/4.04/lib/stringext -I /home/opam/.opam/4.04/lib/typerep -I /home/opam/.opam/4.04/lib/uri -I /home/opam/.opam/4.04/lib/uri-sexp -I /home/opam/.opam/4.04/lib/uri/services -I /home/opam/.opam/4.04/lib/variantslib -I /home/opam/.opam/4.04/lib/yojson -intf-suffix .ml -no-alias-deps -open Oraft__ -o lib/.oraft.objs/byte/oraft__Request_dispatcher.cmo -c -impl lib/request_dispatcher.pp.ml)
# File "lib/request_dispatcher.ml", line 31, characters 14-37:
# Error: Unbound value Stdlib.Hashtbl.find_opt
# Hint: Did you mean find_all?
#     ocamlopt lib/.oraft.objs/native/oraft__Request_dispatcher.{cmx,o} (exit 2)
# (cd _build/default && /home/opam/.opam/4.04/bin/ocamlopt.opt -w -40 -g -I lib/.oraft.objs/byte -I lib/.oraft.objs/native -I /home/opam/.opam/4.04/lib/astring -I /home/opam/.opam/4.04/lib/base -I /home/opam/.opam/4.04/lib/base/caml -I /home/opam/.opam/4.04/lib/base/md5 -I /home/opam/.opam/4.04/lib/base/shadow_stdlib -I /home/opam/.opam/4.04/lib/base64 -I /home/opam/.opam/4.04/lib/bin_prot -I /home/opam/.opam/4.04/lib/bin_prot/shape -I /home/opam/.opam/4.04/lib/biniou -I /home/opam/.opam/4.04/lib/bytes -I /home/opam/.opam/4.04/lib/cohttp -I /home/opam/.opam/4.04/lib/cohttp-lwt -I /home/opam/.opam/4.04/lib/cohttp-lwt-unix -I /home/opam/.opam/4.04/lib/conduit -I /home/opam/.opam/4.04/lib/conduit-lwt -I /home/opam/.opam/4.04/lib/conduit-lwt-unix -I /home/opam/.opam/4.04/lib/core -I /home/opam/.opam/4.04/lib/core_kernel -I /home/opam/.opam/4.04/lib/core_kernel/base_for_tests -I /home/opam/.opam/4.04/lib/domain-name -I /home/opam/.opam/4.04/lib/easy-format -I /home/opam/.opam/4.04/lib/fieldslib -I /home/opam/.opam/4.04/lib/fmt -I /home/opam/.opam/4.04/lib/ipaddr -I /home/opam/.opam/4.04/lib/ipaddr-sexp -I /home/opam/.opam/4.04/lib/ipaddr/unix -I /home/opam/.opam/4.04/lib/jane-street-headers -I /home/opam/.opam/4.04/lib/logs -I /home/opam/.opam/4.04/lib/lwt -I /home/opam/.opam/4.04/lib/lwt/unix -I /home/opam/.opam/4.04/lib/macaddr -I /home/opam/.opam/4.04/lib/magic-mime -I /home/opam/.opam/4.04/lib/mmap -I /home/opam/.opam/4.04/lib/ocaml/threads -I /home/opam/.opam/4.04/lib/ocplib-endian -I /home/opam/.opam/4.04/lib/parsexp -I /home/opam/.opam/4.04/lib/ppx_assert/runtime-lib -I /home/opam/.opam/4.04/lib/ppx_bench/runtime-lib -I /home/opam/.opam/4.04/lib/ppx_compare/runtime-lib -I /home/opam/.opam/4.04/lib/ppx_deriving/runtime -I /home/opam/.opam/4.04/lib/ppx_deriving_yojson/runtime -I /home/opam/.opam/4.04/lib/ppx_expect/collector -I /home/opam/.opam/4.04/lib/ppx_expect/common -I /home/opam/.opam/4.04/lib/ppx_expect/config -I /home/opam/.opam/4.04/lib/ppx_hash/runtime-lib -I /home/opam/.opam/4.04/lib/ppx_inline_test/config -I /home/opam/.opam/4.04/lib/ppx_inline_test/runtime-lib -I /home/opam/.opam/4.04/lib/ppx_sexp_conv/runtime-lib -I /home/opam/.opam/4.04/lib/re -I /home/opam/.opam/4.04/lib/re/posix -I /home/opam/.opam/4.04/lib/result -I /home/opam/.opam/4.04/lib/seq -I /home/opam/.opam/4.04/lib/sexplib -I /home/opam/.opam/4.04/lib/sexplib/unix -I /home/opam/.opam/4.04/lib/sexplib0 -I /home/opam/.opam/4.04/lib/spawn -I /home/opam/.opam/4.04/lib/splittable_random -I /home/opam/.opam/4.04/lib/stdio -I /home/opam/.opam/4.04/lib/stdlib-shims -I /home/opam/.opam/4.04/lib/stringext -I /home/opam/.opam/4.04/lib/typerep -I /home/opam/.opam/4.04/lib/uri -I /home/opam/.opam/4.04/lib/uri-sexp -I /home/opam/.opam/4.04/lib/uri/services -I /home/opam/.opam/4.04/lib/variantslib -I /home/opam/.opam/4.04/lib/yojson -intf-suffix .ml -no-alias-deps -open Oraft__ -o lib/.oraft.objs/native/oraft__Request_dispatcher.cmx -c -impl lib/request_dispatcher.pp.ml)
# File "lib/request_dispatcher.ml", line 31, characters 14-37:
# Error: Unbound value Stdlib.Hashtbl.find_opt
# Hint: Did you mean find_all?
```
So I changed fix the issue in your package description. Could you return the fix upstream or fix it if you want to stay compatible with older versions of the compiler?

**Comment by kit-ty-kate** [comment]
All green. I'll go ahead and merge this. Thanks!

**Comment by komamitsu** [comment]
@kit-ty-kate Thanks a lot!!

**Pull request merged**